src/internal/connector/onedrive/collections.go

```diff
@@ -434,15 +466,62 @@ func (c *Collections) UpdateCollections(
 			// already created and partially populated.
 			updatePath(newPaths, *item.GetId(), folderPath.String())
 
+			if ok {
+				// If we find prev path, that implies these are
+				// moved/renames folders and not new ones
+				prevPath, err := path.FromDataLayerPath(prevColPath, true) // FIXME(meain): converting string to path
+				if err != nil {
+					logger.Ctx(ctx).Errorw("invalid prev path for item", "error", err)
+					return err
+				}
+
+				col, found := c.CollectionMap[folderPath.String()]
+				if !found {
+					col := NewCollection(
+						c.itemClient,
+						folderPath,
+						prevPath,
+						driveID,
+						c.service,
+						c.statusUpdater,
+						c.source,
+						c.ctrl,
+						invalidPrevDelta,
+					)
+					c.CollectionMap[folderPath.String()] = col
+					c.NumContainers++
+				} else {
+					// If we have previously created a collection when
+					// we encountered a file, this gives us a change
+					// to mark the container as moved or not moved
+					// instead of new.
+					ccol, ok := col.(*Collection)
+					if !ok {
+						logger.Ctx(ctx).Errorw("unable to cast onedrive collection", "error", err)
+						return err
+					}
+
+					ccol.SetPreviousPath(prevPath)
+					c.CollectionMap[folderPath.String()] = ccol
+				}
+			}
+
 			if c.source != OneDriveSource {
 				continue
 			}
 
 			fallthrough
 
 		case item.GetFile() != nil:
-			if item.GetDeleted() != nil {
+			if !invalidPrevDelta && item.GetFile() != nil {
+				// Always add a file to the excluded list. If it was
+				// deleted, we want to avoid it. If it was
+				// renamed/moved/modified, we still have to drop the
+				// original one and repull the new one.
```

Comment:
I think this works for now, though we may want to refine it in the future to only add it to the exclude list if the delta was non-empty and valid. Empty deltas mean we're pulling everything and not merging with a base so we don't need to exclude stuff

Reply:
Not sure what you mean by empty delta, but we do have a check to see if we are pulling from a saved delta query or a new one with the `invalidPrevDelta` flag.

Reply:
mkay, that will work as well. Maybe add a comment somewhere that `invalidPrevDelta` also includes an empty delta so that folks know. Or if there's some part of the code that makes that clear then it's fine (I may have forgotten how `invalidPrevDelta`'s value is assigned 😅)

Reply:
It was not properly handling empty case until now, I've added it [here](https://github.com/alcionai/corso/pull/2407/commits/d061fc983ae100f9493626629f52ff62cb7f7415).